dev/tests/integration/testsuite/Magento/Review/_files/different_reviews_different_stores.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+use Magento\TestFramework\Workaround\Override\Fixture\Resolver;
+
+Resolver::getInstance()->requireDataFixture('Magento/Catalog/_files/product_simple.php');
+$store = \Magento\TestFramework\Helper\Bootstrap::getObjectManager()
+                                                ->create(\Magento\Store\Model\Store::class);
+$secondStoreId = $store->load('fixture_second_store', 'code')->getId();
+
+$review = \Magento\TestFramework\Helper\Bootstrap::getObjectManager()->create(
+    \Magento\Review\Model\Review::class,
+    ['data' => ['nickname' => 'Nickname', 'title' => 'Review Summary store 2', 'detail' => 'Review text']]
+);
+$review->setEntityId(
+    $review->getEntityIdByCode(\Magento\Review\Model\Review::ENTITY_PRODUCT_CODE)
+)->setEntityPkValue(
+    1
+)->setStatusId(
+    \Magento\Review\Model\Review::STATUS_PENDING
+)->setStoreId($secondStoreId)->setStores(
+    [
+        $secondStoreId
+    ]
+)->save();
+
+/*
+ * Added a sleep because in a few tests the sql query orders by created at. Without the sleep the reviews
+ * have sometimes the same created at timestamp, that causes this tests randomly to fail.
+ */
+sleep(1);
+
+$review = \Magento\TestFramework\Helper\Bootstrap::getObjectManager()->create(
+    \Magento\Review\Model\Review::class,
+    ['data' => ['nickname' => 'Nickname', 'title' => '2 filter first review store 2', 'detail' => 'Review text']]
+);
+$review->setEntityId(
+    $review->getEntityIdByCode(\Magento\Review\Model\Review::ENTITY_PRODUCT_CODE)
+)->setEntityPkValue(
+    1
+)->setStatusId(
+    \Magento\Review\Model\Review::STATUS_APPROVED
+)->setStoreId($secondStoreId)->setStores(
+    [
+        $secondStoreId
+    ]
+)->save();
+
+/*
+ * Added a sleep because in a few tests the sql query orders by created at. Without the sleep the reviews
+ * have sometimes the same created at timestamp, that causes this tests randomly to fail.
+ */
+sleep(1);
+
+$review = \Magento\TestFramework\Helper\Bootstrap::getObjectManager()->create(
+    \Magento\Review\Model\Review::class,
+    ['data' => ['nickname' => 'Nickname', 'title' => '1 filter second review store 2', 'detail' => 'Review text']]
+);
+$review->setEntityId(
+    $review->getEntityIdByCode(\Magento\Review\Model\Review::ENTITY_PRODUCT_CODE)
+)->setEntityPkValue(
+    1
+)->setStatusId(
+    \Magento\Review\Model\Review::STATUS_APPROVED
+)->setStoreId($secondStoreId)->setStores(
+    [
+        $secondStoreId
+    ]
+)->save();
```

Comment:
Please, do not use deprecated methods. Use a resource model for saving the entity instead. 

Thanks